### PR TITLE
feat(index): add `[contenthash]` support

### PR DESCRIPTION
**Pull request by chenxsan**
### `Issues`

- Fixes https://github.com/webpack-contrib/mini-css-extract-plugin/issues/1

First time doing webpack plugin code :) Hope it won't break anything.

### CODE REVIEW

**Comment by developer-lindner** [comment]
Documentation should be updated as well!

**Comment by vertic4l** [comment]
Let's get this merged. Need it now :P

**Comment by vertic4l** [comment]
@michael-ciniawsky any time table when this gets merged finally?

**Comment by michael-ciniawsky** [comment]
@vertic4l I don't have access here, that's up to Tobias atm :), From my side it's g2g, the only thing notable here is that the hash of the JS chunk will nevertheless change when the CSS changes...

**Comment by michael-ciniawsky** [comment]
So it mitigates caching problems at best (the CSS will keep the hash)

**Comment by vertic4l** [comment]
@michael-ciniawsky well, i can't actually close my own PR within a project (which updates to webpack 4 setup) as long as this plugin or ExtractTextPlugin doens't work properly with incremenatl build, contentHash and webpack 4. Pretty annoying actually, but can't really help either :/

File: src/index.js [hunk 1]
```diff
@@ -95,6 +96,19 @@ class MiniCssExtractPlugin {
   }
 
   apply(compiler) {
+    // add contenthash support
+    compiler.hooks.emit.tap(pluginName, (compilation) => {
+      const regexp = /\[(?:(\w+):)?contenthash(?::([a-z]+\d*))?(?::(\d+))?\]/ig;
+      Object.keys(compilation.assets).forEach((filename) => {
```
Comment (7rulnik) [inline src/index.js]: What about `object.entries`?

Comment (ooflorent) [reply src/index.js]: Not available in node 6.

**Comment by sokra** [comment]
I'll add `[contenthash]` support to the webpack core, as this is also needed for js files when CSS is part of the chunk.

Thanks for your PR anyway.

**Comment by KyleAMathews** [comment]
So would we then want to use `[contenthash]` for js chunks too?

**Pull request closed without merging**